api/src/app.ts

```diff
@@ -152,9 +152,16 @@ export const build = async (
     })
   });
 
-  const provider =
-    EMAIL_PROVIDER === 'ses' ? new SESProvider() : new NodemailerProvider();
-  void fastify.register(mailer, { provider });
+  if (process.env.JEST_WORKER_ID) {
+    const { MockEmailProvider } = await import(
+      './plugins/mail-providers/mock-email-provider'
+    );
+    void fastify.register(mailer, { provider: MockEmailProvider });
+  } else {
+    const provider =
+      EMAIL_PROVIDER === 'ses' ? new SESProvider() : new NodemailerProvider();
+    void fastify.register(mailer, { provider });
+  }
```

Comment:
```suggestion
  const provider =
    EMAIL_PROVIDER === 'ses' ? new SESProvider() : new NodemailerProvider();
  void fastify.register(mailer, { provider });
```
We can handle this all in the test suite, so the core logic doesn't need to care about how we test.